osbs/build/build_request.py

```diff
@@ -377,17 +377,34 @@ def adjust_for_registry_api_versions(self):
     def adjust_for_triggers(self):
         """
         Remove trigger-related plugins if no triggers set
+        import_image plugin is only removed for non-custom base images
+        Triggers are explicitly removed for custom base images
         """
         triggers = self.template['spec'].get('triggers', [])
+
+        remove_plugins = [
+            ("prebuild_plugins", "check_and_set_rebuild"),
+            ("prebuild_plugins", "stop_autorebuild_if_disabled"),
+            ("prebuild_plugins", "update_parent_image_stream_tag"),
+            ("exit_plugins", "sendmail"),
+        ]
+
+        if triggers and self.is_custom_base_image():
+            for when, which in remove_plugins:
+                logger.info("removing %s from request because custom base image",
+                            which)
+                self.dj.remove_plugin(when, which)
+
+            del self.template['spec']['triggers']
+            return
+
         if not triggers:
-            for when, which in [("prebuild_plugins", "check_and_set_rebuild"),
-                                ("prebuild_plugins", "stop_autorebuild_if_disabled"),
-                                ("prebuild_plugins", "update_parent_image_stream_tag"),
-                                ("postbuild_plugins", "import_image"),
-                                ("exit_plugins", "sendmail")]:
+            remove_plugins.append(("postbuild_plugins", "import_image"))
+            for when, which in remove_plugins:
                 logger.info("removing %s from request because there are no triggers",
                             which)
                 self.dj.remove_plugin(when, which)
```

Comment:
I guess these two loops only differ in the reason they log about. Could we avoid repetition by having the `if`s decide on both the list of plugins to remove, and the reason for doing it?